modules/test-renderers.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const nunjucks = require('nunjucks');
+nunjucks.configure('views');
+
+function renderComponentMacro(componentPath, macroName, params) {
+    /**
+     * Mock any global translation strings.
+     * Wherever possible you should avoid requiring context for components
+     */
+    const context = {
+        __(key) {
+            return `[i18n:${key}]`;
+        }
+    };
+
+    const macroParams = JSON.stringify(params, null, 2);
+    const macroString = `{%- from "${componentPath}" import ${macroName} with context -%}{{- ${macroName}(${macroParams}) -}}`;
```

Comment:
This approach is based on https://github.com/alphagov/govuk-frontend/blob/master/lib/jest-helpers.js#L37 so all credit to the GDS team for this approach.